File: src/Client_Java/model/LoginModel.java
package Client_Java.model;

import Client_Java.model.BoggleApp.BoggleClient;
import Client_Java.model.BoggleApp.accountDoesNotExist;
import Client_Java.model.BoggleApp.accountLoggedIn;

public class LoginModel {
    BoggleClient wfImpl;


    public LoginModel(BoggleClient wfImpl) {
        this.wfImpl = wfImpl;
    }

    /**
     * Returns a boolean if the entered account details exists in the database
     * @param username
     * @param password
     * @return
     * @throws Exception
     */
    public String validateAccount(String username, String password)throws accountLoggedIn,accountDoesNotExist {
        try {
            wfImpl.validateAccount(username, password);
            return "valid";
        }catch (Exception e){
            return "Already logged in or does not exist";
        }
    }

    public BoggleClient getWfImpl() {
        return wfImpl;
    }

    public void setWfImpl(BoggleClient wfImpl) {
        this.wfImpl = wfImpl;
    }
}

File: src/Client_Python/src/model/LoginModel.py
from BoggleClient_idl import BoggleClient as BoggleClient
from BoggleClient_idl import accountDoesNotExist
from BoggleClient_idl import accountLoggedIn

class LoginModel:

    """
    This is the constructor
    
    wfmpl variable should only be casted to BoggleClient interface from the BoggleClient_idl (line 142)
    """
    def __init__(self, wfmpl: BoggleClient):
        self.wfmpl = wfmpl

    """
    The validateAccount method will use the validateAccount from the wfmpl object
    """
    def validateAccount(self, username, password):
        try:

            self.wfmpl.validateAccount(username, password)
            return "valid"

        except accountDoesNotExist as adne:
            
            return "Account does not exist"

        except accountLoggedIn as ali:

            return "Account already logged in"

    """
    Returns the wfmpl variable
    """
    def getWfmpl(self):
        return self.wfmpl
    
    """
    Sets the wfmpl variable to a BoggleClient object when used
    """
    def setWfmpl(self, wfmpl: BoggleClient):
        self.wfmpl = wfmpl
